Java/crypto/src/main/java/com/breadwallet/crypto/TransferAttribute.java

```diff
@@ -0,0 +1,20 @@
+package com.breadwallet.crypto;
+
+import com.google.common.base.Optional;
+
+public interface TransferAttribute {
+    String getKey();
+
+    Optional<String> getValue();
+
+    void setValue(Optional<String> value);
+
+    boolean isRequired();
+
+    enum Error {
+        REQUIRED_BUT_NOT_PROVIDED,
+        MISMATHED_TYPE,
+        RELATIONSHIP_INCONSISTENCY
```

Comment:
Are these used somewhere?

Reply:
Utilities::transferAttributeErrorFromCrypto and called on the return from the wallet's validate attribute methods.